### PR TITLE
Refactor qa test scripts

### PR DESCRIPTION
**Pull request by awolant**
## Category:

Refactoring

## Description:

Python testing framework references (which currently is `nosetests`) were spread all over the tests scripts. I refactored the scripts to define the testing framework in one place and reuse this definition across the tests. This will allow to change testing framework easier in the future and adjust the common arguments for all tests in one place.

Additionally, I removed some unused scripts and make targets.

### Affected modules and functionalities:

Most changes are mechanical and can be grouped in three categories:

- Replace manual invocations of test runner with variable, for example: 
`nosetests --verbose test_fw_iterators_detection.py`     
changes to:
`${python_test_runner} ${python_test_args}  test_fw_iterators_detection.py`

- list of pip packages uses single quote rather than double quote:
`pip_packages="pillow torchvision torch"` 
changes to:
`pip_packages='pillow torchvision torch'`
This is needed to delay the evaluation of variables embedded in the list to later, when we know the values

- name of the pip package for testing framework is set once and in the lists for every test variable is used instead:
`pip_packages="nose numpy tensorflow-gpu"`
changes to:
`pip_packages='${python_test_runner_package} numpy tensorflow-gpu'`

Other changes in this PR are: 
- in `test_template.sh` setting proper variables and evaluating `pip_packages` to replace `python_test_runner_package`
- in `test_body.sh` changing the name of function `test_nose` to `test_py_with_framework`
- removing `test_no_setup.sh` files and corresponding make targets as not used and not maintained

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: 2810

### CODE REVIEW

**Comment by awolant** [comment]
!build

**Comment by awolant** [comment]
!build

File: dali/CMakeLists.txt [hunk 1]
```diff
@@ -141,24 +141,6 @@ if (BUILD_PYTHON)
...
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/qa/TL0_python-self-test-readers-decoders/
-    COMMAND PYTHONPATH=${PYTHONPATH} ${PROJECT_SOURCE_DIR}/qa/TL0_python-self-test-readers-decoders/test_no_setup.sh
-  )
-
-  add_custom_command(
-    TARGET check-python
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/qa/TL0_python-self-test-operators/
-    COMMAND PYTHONPATH=${PYTHONPATH} ${PROJECT_SOURCE_DIR}/qa/TL0_python-self-test-operators/test_no_setup.sh
-  )
-
```
Comment (awolant) [inline dali/CMakeLists.txt:161]: Removed as unused

File: qa/TL0_python-self-test-operators/test_no_setup.sh [hunk 2]
```diff
@@ -1,15 +0,0 @@
...
-    return 0
-fi
-
-source test_body.sh
-
-pushd ../../dali/test/python
-
-run_all
-
-popd
```
Comment (awolant) [inline qa/TL0_python-self-test-operators/test_no_setup.sh:15]: Remeved as unused

File: qa/TL0_python-self-test-readers-decoders/test_no_setup.sh [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-    return 0
-fi
-
-source test_body.sh
-
-pushd ../../dali/test/python
-
-run_all
-
-popd
```
Comment (awolant) [inline qa/TL0_python-self-test-readers-decoders/test_no_setup.sh:15]: Removed as unused

File: qa/TL0_python-self-test_xavier/test_no_setup.sh [hunk 4]
```diff
@@ -1,15 +0,0 @@
...
-    return 0
-fi
-
-source test_body.sh
-
-pushd ../../dali/test/python
-
-run_all
-
-popd
```
Comment (awolant) [inline qa/TL0_python-self-test_xavier/test_no_setup.sh:15]: Removed as unused

File: qa/test_template.sh [hunk 5]
```diff
@@ -8,8 +8,13 @@ set -x
...
 source $topdir/qa/setup_test_common.sh
 
+# Set runner for python tests
+python_test_runner_package="nose"
+python_test_runner="python -m nose"
+python_test_args="--verbose -s"
+
 # Set proper CUDA version for packages, like MXNet, requiring it
-pip_packages=$(echo ${pip_packages} | sed "s/##CUDA_VERSION##/${CUDA_VERSION}/")
+pip_packages=$(eval "echo \"${pip_packages}\"" | sed "s/##CUDA_VERSION##/${CUDA_VERSION}/")
```
Comment (awolant) [inline qa/test_template.sh:17]: This change allows to evaluate `pip_packages` and replace placeholders with current values. 

For example:

`${python_test_runner_package} numpy>=1.17 numba scipy librosa==0.8.1` 

becomes:

`nose numpy>=1.17 numba scipy librosa==0.8.1`

File: qa/test_template.sh [hunk 6]
```diff
@@ -8,8 +8,13 @@ set -x
 topdir=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )/..
 source $topdir/qa/setup_test_common.sh
 
+# Set runner for python tests
+python_test_runner_package="nose"
+python_test_runner="python -m nose"
+python_test_args="--verbose -s"
+
```
Comment (awolant) [inline qa/test_template.sh:15]: Setting Python testing framework. First is the name of pip package to install, second is a command to run the test, third are args that we want to pass to all tests.

**Comment by awolant** [comment]
!build

**Comment by awolant** [comment]
!build

**Comment by awolant** [comment]
!build

File: qa/TL0_FW_iterators/test_mxnet.sh [hunk 7]
```diff
@@ -19,9 +19,9 @@ test_body() {
                 --workers 3 --prefetch 2 -i 2 --epochs 2 --fp16
         done
     fi
-    nosetests --verbose -m '(?:^|[\b_\./-])[Tt]est.*mxnet*' test_fw_iterators_detection.py
-    nosetests --verbose -m '(?:^|[\b_\./-])[Tt]est.*mxnet*' test_fw_iterators.py
-    nosetests --verbose -m '(?:^|[\b_\./-])[Tt]est.*gluon*' test_fw_iterators.py
+    ${python_test_runner} ${python_test_args} -m '(?:^|[\b_\./-])[Tt]est.*mxnet*' test_fw_iterators_detection.py
```
Comment (JanuszL) [inline qa/TL0_FW_iterators/test_mxnet.sh:22]: This looks lengthy. How about `${python_invoke_test}` and `python_invoke_test='${python_test_runner} ${python_test_args}'`.

**Review by banasraf — APPROVED** [review]

File: qa/TL0_FW_iterators/test_mxnet.sh (hunk 7 above)
Comment (awolant) [reply qa/TL0_FW_iterators/test_mxnet.sh]: Done

**Comment by awolant** [comment]
!build

**Review by JanuszL — APPROVED** [review]

**Comment by awolant** [comment]
!build

**Comment by awolant** [comment]
!build

**Pull request merged**